tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1454,6 +1454,32 @@ Here are some examples of expressions::
 Identifiers, attributes, and subscripts are valid assignment targets. Other expressions cannot be on the left side of
 an assignment.
 
+self
+^^^^
+
+``self`` can be used to refer to the current instance and is often equivalent to directly referring to symbols available in
+the current script, however, while this is generally considered a bad practice, ``self`` also allows you to access properties,
+methods, and other names that are defined dynamically (i.e. are expected to exist in subtypes of the current script, or are
+provided using :ref:`_get<class_Object_private_method__get>`).
+
+::
+
+    class_name Weapon extends Node
+
+    func _ready() -> void:
+        print(self.damage)
+
+``damage`` is not defined in ``Weapon``, but using ``self`` we can assume that it will exist at runtime. For example:
+
+::
+
+    class_name Pistol extends Weapon
```

Comment:
In my opinion, it would be better to separate the example with dynamic properties and inheritance, this section may be difficult for a beginner to understand.

Reply:
There are examples in `Object._get()`, `Object._set()` and related methods' docs already. I don't think it would be beneficial to duplicate them here?